plugins/modules/foreman_operating_system.py

```diff
@@ -216,6 +216,14 @@ def main():
             )),
             state=dict(default='present', choices=['present', 'present_with_defaults', 'absent']),
         ),
+        required_if=[
+            ['state', 'present', ['name', 'major', 'family']],
+            ['state', 'present_with_defaults', ['name', 'major', 'family']],
+        ],
+        required_one_of=[
```

Comment:
just so that I parse it right. you require name, major, family for present/with_defaults as that's what we need for creation, but for absent, we can either go by name+major OR description. right?

Reply:
talking to myself, is there a way to better express that in the docs?

Reply:
that _is_ the intention.